tests/src/main/java/com/hazelcast/simulator/tests/synthetic/SyntheticTest.java

```diff
@@ -166,69 +161,46 @@ public void run() {
             }
         }
 
-        public void doRun() throws Exception {
-            long iteration = 0;
-
-            while (!context.isStopped()) {
-
-                int partitionId = nextPartitionId();
-
-                ICompletableFuture f = invoke(partitionId);
-                //latency.started();
-
-                if (syncInvocation) {
-                    if (syncFrequency == 1) {
-                        f.get();
-                    } else {
-                        if (iteration > 0 && iteration % syncFrequency == 0) {
-                            for (ICompletableFuture future : futureList) {
-                                future.get();
-                            }
-                            futureList.clear();
-                        } else {
-                            futureList.add(f);
+        private void timeStep() throws Exception {
+            ICompletableFuture<Object> future = invokeOnNextPartition();
+            if (syncInvocation) {
+                intervalProbe.started();
+                if (syncFrequency == 1) {
+                    future.get();
+                } else {
+                    futureList.add(future);
```

Comment:
We always have to add the future to the list in this code branch. In the old version it got lost if the future list was read.